context/src/main/java/io/micronaut/runtime/server/GracefulShutdownCapable.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.runtime.server;
```

Comment:
should this be related to the server? Graceful shutdown may not have a server, like in the case of a headless Kafka listener.

Reply:
I'll make a runtime.graceful subpackage instead